.github/workflows/test-docker.yml

```diff
@@ -37,10 +37,25 @@ jobs:
         with:
           name: ${{ env.RUN_STAMP }}-${{ inputs.build-name }}
 
+      - name: Choose Dockerfiles
+        if: inputs.name == 'clang-cpu'
+        run: |
+          echo Additional_Dockerfile=docker/Dockerfile.clang >>$GITHUB_ENV
+     
+      - name: Choose Dockerfiles
+        if: inputs.name == 'cuda'
+        run: |
+          echo Additional_Dockerfile=docker/Dockerfile.cuda >>$GITHUB_ENV
+      
+      - name: Choose Dockerfiles
+        if: inputs.name == 'l0'
+        run: |
+          echo Additional_Dockerfile=docker/Dockerfile.l0 >>$GITHUB_ENV
+
```

Comment:
I believe this overcomplicates things; we always can name the dockerfile as we like, so there is no need to have this additional "switch" to choose the name